install/kubernetes/cilium/values.yaml

```diff
@@ -1253,6 +1257,19 @@ tls:
   enabled: true
   secretsBackend: local
 
+  # -- Deprecated in favor of tls.ca. To be removed in 1.13.
+  # base64 encoded PEM values for the CA certificate and private key.
+  # This can be used as common CA to generate certificates used by hubble and clustermesh components
+  ca:
```

Comment:
I'm still a bit weary on adding this to the global `tls` section, as that section is currently used to configure the secrets backend of https://docs.cilium.io/en/v1.11/gettingstarted/tls-visibility/

I really think we need to be better at documenting the difference between the two, because users will confuse the Hubble API mTLS protection with the TLS visibility feature. It doesn't help that both fields are mainly used for network visibility via Hubble, but in a very different way (one defines the certificates used to protect the internal Hubble API with mTLS, the other configures how certificates and secret keys are loaded to decrypt traffic to be observed by Hubble).

I would propose the following here:
1. Remove the `tls.enabled` flag (it's currently a no-op and might confuse users who know about `hubble.tls.enabled`).
2. Add some documentation to `tls.secretsBackend`. For example something along the lines of:
> This configures how the Cilium agent loads the secrets used TLS-aware CiliumNetworkPolicies (namely the secrets referenced by terminatingTLS and originatingTLS).
> Possible values:
 > - local 
 > - tls

Reply:
thanks for your super detailed comment here :100: